### PR TITLE
[3.0] Fix an issue with mediumtext type

### LINKED ISSUE #8126: [3.0]: Adding columns with type = 'mediumtext' does not work
### Basic Information

This script fine works in 2.1, but doesn't work in 3.0.

It must add two tables - lp_page_tags, and lp_pages, but adds only lp_page_tags, for unknown reasons.

### Steps to reproduce

1. Try to run this database.php:
```php
<?php

if (file_exists(__DIR__ . '/SSI.php') && ! defined('SMF')) {
	require_once __DIR__ . '/SSI.php';
} elseif (! defined('SMF')) {
	die('<b>Error:</b> Cannot install - please verify that you put this file in the same place as SMF\'s index.php and SSI.php files.');
}

if ((SMF === 'SSI') && ! $user_info['is_admin']) {
	die('Admin privileges required.');
}

$tables = [];

$tables[] = [
	'name'    => 'lp_page_tags',
	'columns' => [
		[
			'name'     => 'page_id',
			'type'     => 'int',
			'size'     => 10,
			'unsigned' => true
		],
		[
			'name'     => 'tag_id',
			'type'     => 'int',
			'size'     => 10,
			'unsigned' => true
		]
	],
	'indexes' => [
		[
			'type'    => 'primary',
			'columns' => ['page_id', 'tag_id']
		]
	]
];

$tables[] = [
	'name' => 'lp_pages',
	'columns' => [
		[
			'name'     => 'page_id',
			'type'     => 'smallint',
			'size'     => 5,
			'unsigned' => true,
			'auto'     => true
		],
		[
			'name' => 'description',
			'type' => 'varchar',
			'size' => 255,
			'null' => true
		],
		[
			'name' => 'content',
			'type' => 'mediumtext',
			'null' => false
		],
		[
			'name'    => 'type',
			'type'    => 'varchar',
			'size'    => 10,
			'default' => 'bbc',
			'null'    => false
		],	
	],
	'indexes' => [
		[
			'type'    => 'primary',
			'columns' => ['page_id']
		],
		[
			'type'    => 'unique',
			'columns' => ['alias']
		]
	],
];

db_extend('packages');

foreach ($tables as $table) {
	$smcFunc['db_create_table']('{db_prefix}' . $table['name'], $table['columns'], $table['indexes']);
}

if (SMF === 'SSI') {
	echo 'Database changes are complete! Please wait...';
}

```

### Expected result

Two tables in the database: `lp_page_tags`, `lp_pages`

### Actual result

One table in the database: `lp_page_tags`

### Version/Git revision

d594310

### Database Engine

MySQL

### Database Version

10.11.6-MariaDB

### PHP Version

8.2.16

### Logs

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by dragomano**
Fixes #8126

### CODE REVIEW

File: Sources/Db/APIs/MySQL.php [hunk 1]
```diff
@@ -1349,6 +1349,8 @@ public function calculate_type(string $type_name, ?int $type_size = null, bool $
 			$types = [
 				'inet' => 'varbinary',
 				'uuid' => 'binary',
+				'mediumtext' => 'text',
+				'largetext' => 'text',
```
Comment (Sesquipedalian) [inline Sources/Db/APIs/MySQL.php]: This doesn't seem correct to me. Why would we want to change MEDIUMTEXT and LARGETEXT data types to regular TEXT?

**Review by Sesquipedalian — CHANGES_REQUESTED** [review]

File: Sources/Db/APIs/MySQL.php (hunk 1 above)
Comment (dragomano) [reply Sources/Db/APIs/MySQL.php]: You're right. This means something changed somewhere, because in Alpha 1 `database.php` worked as expected.

Comment (dragomano) [reply Sources/Db/APIs/MySQL.php]: Now a query like this is generated:
```sql
CREATE TABLE `smf_alpha`.smf_lp_pages ( `page_id` smallint(5) unsigned auto_increment, `alias` varchar(255) NOT NULL , `description` varchar(255) , `content` mediumtext(0) NOT NULL , `type` varchar(10) NOT NULL DEFAULT 'bbc', PRIMARY KEY (page_id), UNIQUE alias (alias(191))) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_general_ci
```
And it needs to be like this:
```sql
CREATE TABLE `smf_alpha`.smf_lp_pages ( `page_id` smallint(5) unsigned auto_increment, `alias` varchar(255) NOT NULL , `description` varchar(255) , `content` mediumtext NOT NULL , `type` varchar(10) NOT NULL DEFAULT 'bbc', PRIMARY KEY (page_id), UNIQUE alias (alias(191))) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_general_ci
```

Comment (jdarwood007) [reply Sources/Db/APIs/MySQL.php]: type hinting issue is defaulting the size to 0 and it needs to be nullable so when null we don't pass a size.

**Review by Sesquipedalian — APPROVED** [review]

**Pull request merged**